### DISCUSSION TITLE
How to make a table (with sub-diagram on cell) with PlantUML?

### DISCUSSION THREAD

**Discussion by The-Lum** [plantuml/plantuml#1157] [Q&A]
Hello all,

_As the title:_
- How to make a table (with sub-diagram on cell) with PlantUML?

💡 Every ideas are welcome.
_[All but not `creole table` (where sub-diagram is very difficult to manage)]_

**Comment by The-Lum** [comment]
In fact, the main question is actually:
- How to easily include **sub-diagrams** in _[creole]_ table ?

_[@arnaudroques: this would be very useful for structuring the plantuml documentation and the help pages]_

**Reply by arnaudroques** [reply]
Maybe we should allow a [similar table syntax as in AsciiDoc](https://docs.asciidoctor.org/asciidoc/latest/tables/build-a-basic-table/)?

Or we could also copy `<table>` `<tr>` `<td>` syntax from HTML ?

Which one sounds better?
Or we could allow **both** syntaxes ... :-)

**Reply by The-Lum** [reply]
As you want.
_I don't know what will be the best...._

But the requirements are:
- allow sub-diagram
- allow `|`, `;`, `}` within
- allow multi-line
- allow `<code></code>`
- _why not_ cell merging/fusing...

in order to solve, for example:
- https://forum.plantuml.net/10521/embedded-code-is-not-parsed-within-table-cells
- ...

Regards.

**Reply by The-Lum** [reply]
Hello all,

- _Here is a big question..._

Compare _(on a creole context...)_:
- AsciiDoc

```AsciiDoc
[cols="1,1"]
|===
|Cell in column 1, row 1 with <code>
@startebnf
title Test of EBNF alt
alt = a 
| b
    | c;
@endebnf
</code>
|Cell in column 2, row 1 with result
{{ebnf
title Test of EBNF alt
alt = a 
| b
    | c;
}}
|=== 
```
- HTML

```html
<table border=2>
  <tr>
    <td>Cell in column 1, row 1 with
    <code>
    @startebnf
    title Test of EBNF alt
    alt = a 
    | b
        | c;
    @endebnf
    </code>
    </td>
    <td>Cell in column 2, row 1 with result
    {{ebnf
    title Test of EBNF alt
    alt = a 
    | b
        | c;
    }}
    </td>
  </tr>
</table>
```

- What are the best?

_No idea IMHO..._
Regards.

**Comment by Potherca** [comment]
I would vote for the HTML version, as that seems less esoteric (i.e. easier to understand)
